spec/abstract/RISCV64/Arch_Structs_A.thy

```diff
@@ -222,6 +222,11 @@ definition max_pt_level :: vm_level
   where
   "max_pt_level = asid_pool_level - 1"
 
+lemma vm_level_exhaust:
+  fixes n :: vm_level
+  shows "\<lbrakk> n = 0 \<Longrightarrow> Q; n = 1 \<Longrightarrow> Q; n = 2 \<Longrightarrow> Q; n = 3 \<Longrightarrow> Q \<rbrakk> \<Longrightarrow> Q"
+  by (induct n; clarsimp)
```

Comment:
This lemma is problematic, because the value of `vm_levels` might change in the future (both up and down) based on configuration choices. The reason we don't have one like it is because we explicitly want proofs to reason about levels generically (apart from the existence of the first two levels and the existence of a bottom level).

It makes sense to use in concrete example states, but otherwise we should not use it and not make it available here.

Reply:
Any idea what lemmas I should be using instead? I run into annoying subgoals that amount to:
`⟦level < 2; level ≠ 0; level ≠ 1 ⟧ ⟹ False"`

Reply:
That lemma above would be fine, we are guaranteed to have at least 2 levels. What we don't want is a lemma that is equivalent to `vm_levels` having a specific value.

Reply:
(Actually that lemma would even be true for `vm_levels < 2`, so it's definitely fine).